realm/realm-jni/src/io_realm_internal_tableview.cpp

```diff
@@ -165,9 +170,11 @@ JNIEXPORT jlong JNICALL Java_io_realm_internal_TableView_nativeGetSourceRowIndex
 {
     try {
         if (!VIEW_VALID_AND_IN_SYNC(env, nativeViewPtr))
-            return npos;
+            return -1;
         if (!ROW_INDEX_VALID(env, TV(nativeViewPtr), rowIndex))
-            return npos;
+            return -1;
+        if (!TV(nativeViewPtr)->is_row_attached(rowIndex))
+            return -1;
     } CATCH_STD()
     return TV(nativeViewPtr)->get_source_ndx(S(rowIndex));   // noexcept
```

Comment:
This is the pattern we use elsewhere, also `get_source_ndx` is marked as noexcept.
I don't mind changing it, I just assumed there was a reason our `*_VALID` methods are inside a `try/catch`?